dispatcher.go

```diff
@@ -45,6 +46,7 @@ func NewDispatcher(globals map[string]interface{}, cfg []config.Node) (*Dispatch
 	parsedCfg, err := parseDispatcherRootCfg(globals, cfg)
 	return &Dispatcher{
 		dispatcherCfg: parsedCfg,
+		Log: log.DefaultLogger,
```

Comment:
Here is the problem: this Logger is replaced by module that embed dispatcher functionality.
This means that it will be replaced with module with possibly nil Out again.

As I said in https://github.com/foxcpp/maddy/pull/124/commits/35c1cf5b2a7ea0b6ed81cabad10187739ab012db#r319509126, it is enough to have fixed deliveryLogger version that uses log.DefaultLogger.Out if Loggers Out is nil. log.DefaultLogger.Out is always non-nil.